app/forms/image-upload.tsx

```diff
@@ -640,3 +655,135 @@ export function CreateImageSideModalForm() {
+  efiPartOffset,
+  isBootableCd,
+  isCompressed,
+}: {
+  efiPartOffset: number
+  isBootableCd: boolean
+  isCompressed: boolean
+}) {
+  // this message should only appear if the image doesn't have a header
+  // marker we are looking for and does not appear to be compressed
+  const efiPartOrBootable = efiPartOffset !== -1 || isBootableCd
+  if (efiPartOrBootable && !isCompressed) return null
+
+  const content = (
+    <div className="flex flex-col space-y-2">
+      <ul className="ml-4 list-disc">
+        {!efiPartOrBootable && (
+          <li>
+            <div>Bootable markers not found at any block size.</div>
+            <div>
+              Expected either “EFI PART” marker at offsets 512 / 2048 / 4096 or “CD001” at
+              offset 0x8001 (for a bootable CD).
+            </div>
+          </li>
+        )}
+        {isCompressed && (
+          <li>
+            <div>This might be a compressed image.</div>
+            <div>
+              Only raw, uncompressed images are supported. Files such as qcow2, vmdk,
+              img.gz, iso.7z may not work.
+            </div>
+          </li>
+        )}
+      </ul>
+      <div>
+        Learn more about{' '}
+        <a
+          target="_blank"
+          rel="noreferrer"
+          href={links.preparingImagesDocs}
+          className="inline-flex items-center underline"
+        >
+          preparing images for import
+          <OpenLink12Icon className="ml-1" />
+        </a>
+      </div>
+    </div>
+  )
+
+  return (
+    <Message
+      variant="info"
+      title="This image might not be bootable"
+      className="[&>*]:space-y-2"
+      content={content}
+    />
+  )
+}
+
+async function readAtOffset(file: File, offset: number, length: number) {
+  const reader = new FileReader()
+
+  const promise = new Promise<string | undefined>((resolve, reject) => {
+    reader.onloadend = (e) => {
+      if (
+        e.target?.readyState === FileReader.DONE &&
+        // should always be true because we're using readAsArrayBuffer
+        e.target.result instanceof ArrayBuffer
+      ) {
+        resolve(String.fromCharCode(...new Uint8Array(e.target.result)))
+        return
+      }
+      resolve(undefined)
+    }
+
+    reader.onerror = (error) => {
+      console.error(`Error reading file at offset ${offset}:`, error)
+      reject(error)
+    }
+  })
+
+  reader.readAsArrayBuffer(file.slice(offset, offset + length))
+  return promise
+}
+
+async function getEfiPartOffset(file: File) {
+  const offsets = [512, 2048, 4096]
+  for (const offset of offsets) {
+    const isMatch = (await readAtOffset(file, offset, 8)) === 'EFI PART'
+    if (isMatch) return offset
+  }
+  return -1
+}
+
+const compressedExts = ['.gz', '.7z', '.qcow2', '.vmdk']
+
+const validateImage = async (file: File) => {
+  const lowerFileName = file.name.toLowerCase()
+  return {
+    efiPartOffset: await getEfiPartOffset(file),
+    isBootableCd: (await readAtOffset(file, 0x8001, 5)) === 'CD001',
```

Comment:
So this doesn't quite implement the decision tree of "if no `EFI PART` found, look for `CD001`. This is likely problematic in terms of what we tell customers for bootable hybrid GPT+ISO, which a lot of Linux distro ISOs are so that you can `dd` them directly to a flash drive (512-byte blocks!) and have them still boot. In this particular case, I think it's better to prefer the GPT and tell the user they should use 512-byte blocks.

My expectation is that one of these would return `{efiPartOffset: 512, isBootableCd: true, ...}` — would that always show a "this image might not be bootable" message of one kind or another?